platform/ios/src/MGLMapView.mm

```diff
@@ -4733,6 +4757,57 @@ - (void)updateAnnotationViews
     [CATransaction commit];
 }
 
+- (void)updateCalloutView
+{
+    UIView <MGLCalloutView> *calloutView = self.calloutViewForSelectedAnnotation;
+    id <MGLAnnotation> annotation = calloutView.representedObject;
+    
+    if (calloutView && annotation)
+    {
+        MGLAnnotationTag tag = [self annotationTagForAnnotation:annotation];
+        MGLAnnotationContext &annotationContext = _annotationContextsByAnnotationTag.at(tag);
+        MGLAnnotationView *annotationView = annotationContext.annotationView;
+        
+        // Update callout view for view annotation
+        if (annotationView)
+        {
+            CGPoint point = annotationView.center;
+            point.x += annotationView.centerOffset.dx;
+            point.y += annotationView.centerOffset.dy;
+            point.y -= CGRectGetMidY(annotationView.bounds);
+            
+            if ( ! CGPointEqualToPoint(calloutView.center, point))
+            {
+                calloutView.center = point;
+            }
+        }
+        else // Update callout view for gl annotation
+        {
+            MGLAnnotationImage *image = [self imageOfAnnotationWithTag:tag];
+            
+            if (!image && [self.delegate respondsToSelector:@selector(mapView:imageForAnnotation:)]) {
+                image = [self.delegate mapView:self imageForAnnotation:annotation];
+            }
+            
+            if (!image)
+            {
+                image = [self dequeueReusableAnnotationImageWithIdentifier:MGLDefaultStyleMarkerSymbolName];
+            }
+            
+            if (image)
+            {
+                CGRect rect = [self positioningRectForCalloutForAnnotationWithTag:tag];
+                CGRect insetRect = UIEdgeInsetsInsetRect(rect, image.image.alignmentRectInsets);
```

Comment:
`-positioningRectForCalloutForAnnotationWithTag:` already calls `-frameOfImage:centeredAtCoordinate:`, so it already accounts for the alignment rect insets.